### ISSUE TITLE
implement ft_statfun_bayesfactor

### ISSUE THREAD

**Issue by robertoostenveld** [fieldtrip/fieldtrip#1294]
Following the discussion at the MEG meeting, I was wondering whether it would make sense to implement `ft_statfun_bayesfactor` similar to `ft_statfun_cohensd` as demonstrated on the [effectsize](http://www.fieldtriptoolbox.org/example/effectsize/) example.  

It would be very little work (like ~10 minutes) if we implement it op top of https://nl.mathworks.com/matlabcentral/fileexchange/69794-bayesfactor (the first hit in google) or similar code, which we can include in `fieldtrip/external`.

@palday and Emma, what would you think of that?

**Comment by palday** [comment]
Bayes factors are only really meaningful if there are decent priors. While I like the idea of having more support for Bayesian stats, I'm a little bit nervous about having a Bayes factor without some decent prior recommendations. I'm also a little bit uncertain whether it makes to have a Bayes factor conditioned on a cluster test. I've been thinking a lot about potentially using Bayes factors as a way to avoid cluster-based permutation tests but don't have my thoughts in an usable form yet.  @behinger might have some opinions a swell.

tl;dr: Implement it, but maybe don't advertise it just yet until we have thought some more about good recommendations.

**Comment by robertoostenveld** [comment]
This would not be conditioned on a cluster test, the statfun is a general mechanism and also used for descriptive measures, not only for inference. But I agree that its value would be limited (but that is with all fancy methods that we implement in FT, most of them are only as valuable as the researcher's understanding of them). And I don't see an easy way to implement it without anything else than flat priors. 

If we have it implemented, it becomes easier to concretely discuss how we could put it to use. E.g. I would be interested in seeing how http://www.fieldtriptoolbox.org/assets/img/example/effectsize/figure2.png would look like with BF plotted as function of channel\*time instead of cohensd. And more importantly, how the reasoning at the end of the example changes.

**Comment by behinger** [comment]
For single time-point BF, I have seen it done once (https://www.ncbi.nlm.nih.gov/pubmed/31260837) (but havent looked for it). Some people claim that default priors BFs for standardized variables might make sense - but then more or less always they look at multiple priors, which makes visualization in multi-dimensional spaces difficult again. Equivalence testing (ala Lakens et al) could be a potential candidate to implement as well.


Wrote this before roberts response: BF for cluster test, difficult.  so far, but I remain skeptical. Generally, I am very skeptical to extract any meaningfull statistical information from significant clusters, as they are biased by definition (even if you take the "rectangle" solution you proposed, the onset/offset of a cluster is still biased, Sassenhagen 2019 & Fieldtrip FAQ)).

**Comment by Spaak** [comment]
Relevant are this: https://github.com/Spaak/contextual-cueing-meg/blob/master/matlab/lib/thirdparty/t1smpbf.m for a standard BF for a one- or paired-sample t-test and this: https://github.com/Spaak/contextual-cueing-meg/blob/master/matlab/lib/thirdparty/t2smpbf.m for an independent samples t-test.

Source for the approach is the following paper: Rouder JN, Speckman PL, Sun D, Morey RD, Iverson G (2009) Bayesian t tests for accepting and rejecting the null hypothesis. Psychonomic Bulletin & Review 16:225–237.

Credits for the code go to Eric-Jan Wagemakers and Sam Schwarzkopf at UCL, who ported this to Matlab. I can't quite remember where I originally found this code, but after some googleing I found the exact same functions here: https://figshare.com/articles/Bayes_Factors_Matlab_functions/1357917 which is how I know where to attribute credit.
